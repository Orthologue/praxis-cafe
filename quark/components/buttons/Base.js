// external imports
import React from 'react'
import { Text, StyleSheet, Animated, Easing, TouchableWithoutFeedback } from 'react-native'
import PropTypes from 'prop-types'
import { baseDim } from '../../../quark'
// local imports
import { styles, containerSizes, textSizes } from './styles'


class BaseButton extends React.Component {
    _pressIn() {
        this.setState({
            animation: Animated.timing(this.state.opacity, {
                toValue: 1,
                duration: 125,
            })
        }, () => {
            // start the animation
            this.state.animation.start()
        })
    }

    _pressOut(){
        // stop the keypress animation if its running
        this.state.animation.stop()

        // start the animation to return to normal state
        Animated.timing(this.state.opacity, {
            toValue: 0,
            duration: 150,
        }).start()
    }

    constructor(...args) {
        super(...args)

        // initial state
        this.state = {
            opacity: new Animated.Value(0),
        }

        // function binds
        this._pressIn = this._pressIn.bind(this)
        this._pressOut = this._pressOut.bind(this)
    }

    render() {
        const {style, children, size, activeColor, defaultColor, textColor, ...unused} = this.props

        return (
            <TouchableWithoutFeedback onPressIn={this._pressIn} onPressOut={this._pressOut}>
                <Animated.View
                    {...unused}
                    style={[
                        styles.container,
                        containerSizes[size],
                        {
                            backgroundColor: this.state.opacity.interpolate({
                                inputRange: [0, 1],
                                outputRange: [defaultColor, activeColor]
                            }),
                        },
                        style,
                    ]}
                >
                    {children}
                </Animated.View>
            </TouchableWithoutFeedback>
        )
    }
}

export const buttonDefaultProps = {
    size: "medium",
}

export const buttonPropTypes = {
    size: PropTypes.string,
}

BaseButton.defaultProps = {
    ...buttonDefaultProps,
}

BaseButton.propTypes = {
    ...buttonPropTypes,
    defaultColor: PropTypes.string,
    activeColor: PropTypes.string,
    hoverColor: PropTypes.string,
    textColor: PropTypes.string,
}


export default BaseButton
