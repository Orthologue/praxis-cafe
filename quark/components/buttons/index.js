export BaseButton from './Base'
export Button from './Primary'
export Warning from './Warning'
