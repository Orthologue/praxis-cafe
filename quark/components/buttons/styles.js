// external imports
import { StyleSheet } from 'react-native'
// local imports
import { baseDim } from '../../styles'

export const containerSizes = StyleSheet.create({
    small: {
       height: 7 * baseDim,
    },
    medium: {
       height: 10 * baseDim,
    },
    large: {
        height: 14 * baseDim,
    },
})

export const textSizes = StyleSheet.create({
    small: {
        fontSize: 14,
    },
    medium: {
        fontSize: 16,
    },
    large: {
        fontSize: 18,
    },
})

export const styles = StyleSheet.create({
    container: {
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 2,
        overflow: 'hidden',
    },
    underlay: {
    }
})
