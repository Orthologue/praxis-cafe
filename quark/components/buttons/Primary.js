// external imports
import React from 'react'
// local imports
import { primaryColor, primaryColorDark } from '../../styles'
import BaseButton from './Base'

const Button = ({...unused}) => (
    <BaseButton
        defaultColor={primaryColor}
        activeColor={primaryColorDark}
        textColor="white"
        {...unused}
    />
)

export default Button
