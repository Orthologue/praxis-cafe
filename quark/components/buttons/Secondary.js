// external imports
import React from 'react'
// local imports
import { grey5, primaryColor } from '../../styles'
import BaseWithText from './Text'

const SecondaryButton = ({...unused}) => (
    <BaseWithText
        defaultColor="white"
        activeColor="white"
        textColor={grey5}
        {...unused}
    />
)

export default SecondaryButton
