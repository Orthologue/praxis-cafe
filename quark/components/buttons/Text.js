// external imports
import React from 'react'
import { Text } from 'react-native'
// local imports
import Base from './Base'
import {textSizes} from './styles'

const ButtonWithText = ({size, textColor, children, ...unused}) => (
    <Base size={size} {...unused}>
        <Text
            style={[
                textSizes[size],
                { color: textColor }
            ]}
        >
            {children}
        </Text>
    </Base>
)

ButtonWithText.defaultProps = {
    size: "medium",
}


export default ButtonWithText
