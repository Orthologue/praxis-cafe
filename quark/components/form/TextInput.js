// external imports
import React, { Component } from 'react'
import { TextInput as NativeInput, View, Text, StyleSheet, Modal } from 'react-native'
// local imports
import { primaryColor, secondaryColor, grey2, grey4, baseDim } from '../../styles'
import { Button, PrimaryButton, SecondaryButton } from '../buttons'
import { StatusBar } from '../../native'

class TextInput extends Component{

    state = {
        showModal: false,
        value: '',
    }

    _showModal() {
        this.setState({
            showModal: true,
        })
    }

    _cancel() {
        this.setState({
            showModal: false,
        })
    }

    _submit() {
        this.setState({
            showModal: false,
        })
    }

    constructor(...args) {
        super(...args)

        // function binds
        this._showModal = this._showModal.bind(this)
        this._cancel = this._cancel.bind(this)
        this._submit = this._submit.bind(this)
    }

    componentWillReceiveProps(newProps) {
        console.log(newProps.children)
    }

    render() {
        const {style, children, content, onChange, ...unused} = this.props

        return (
            <View style={[styles.container, style]}>
                <Button
                    style={[
                        styles.inputPlaceholder,
                        styles.inactive,
                    ]}
                    onPress={this._showModal}
                >
                    {this.state.value}
                </Button>
                    {this.state.showModal && (
                        <Modal style={{backgroundColor: "white"}}>
                            <StatusBar/>
                            <View style={styles.content}>
                                <NativeInput
                                    autoFocus
                                    style={styles.input}
                                    selectionColor={primaryColor}
                                    returnKeyType="done"
                                    onSubmitEditing={this._submit}
                                    value={this.state.value}
                                    onChangeText={value => this.setState({ value })}
                                />
                                <View style={styles.modalControls}>
                                    <SecondaryButton onPress={this._cancel} style={[styles.button, styles.leftButton]}>
                                        Cancel
                                    </SecondaryButton>
                                    <PrimaryButton onPress={this._submit} style={[styles.button, styles.rightButton]}>
                                        Submit
                                    </PrimaryButton>
                                </View>
                            </View>
                        </Modal>
                    )}
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    inputPlaceholder: {
        borderRadius: 2,
        borderWidth: 1,
        flex: 1,
        alignItems: 'flex-start',
        justifyContent: 'center',
        height: 10 * baseDim,
        paddingTop: 3 * baseDim,
        paddingBottom: 3 * baseDim,
        paddingLeft: 2 * baseDim,
        paddingRight: 2 * baseDim,
    },
    inactive: {
        borderColor: grey2,
    },
    active: {

    },
    disabled: {

    },
    modalControls: {
        display: 'flex',
        flexDirection: 'row',
    },
    input: {
        fontSize: 12 * baseDim,
        width: '60%',
        height: 15 * baseDim,
        display: 'flex',
        alignSelf: 'center',
        textAlign: 'center',
        marginTop: 8 * baseDim,
        marginBottom: 8 * baseDim,
    },
    content: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
    },
    button: {
        width: 100,
    },
    leftButton: {
        marginRight: 3 * baseDim,
    },
    rightButton: {

    },
})

export default TextInput
