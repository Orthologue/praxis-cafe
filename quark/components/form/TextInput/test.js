// external imports
import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import ReactTestUtils from 'react-addons-test-utils'
// local imports
import TextInput from '.'
import { Button } from '../../buttons'

describe('Form', () => {
    describe('TextInput', () => {
        test('mounts a placeholder without content', () => {
            // mount an input to test
            const wrapper = renderer.create(
                <TextInput />
            ).toJSON()

            expect(wrapper).toMatchSnapshot()
        })

        test('renders content', () => {
            // mount an input to test
            const wrapper = renderer.create(
                <TextInput>
                    hello
                </TextInput>
            ).toJSON()

            expect(wrapper).toMatchSnapshot()
        })
    })
})
