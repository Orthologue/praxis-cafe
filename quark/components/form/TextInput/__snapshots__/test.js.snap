// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form TextInput mounts a placeholder without content 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderColor": "#EAEAEE",
        "borderRadius": 2,
        "borderWidth": 1,
        "flex": 1,
        "height": 40,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#48484D",
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Form TextInput renders content 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderColor": "#EAEAEE",
        "borderRadius": 2,
        "borderWidth": 1,
        "flex": 1,
        "height": 40,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "color": "#48484D",
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
